userspace/libscap/engine/gvisor/scap_gvisor.cpp

```diff
@@ -398,18 +398,18 @@ int32_t engine::stop_capture()
 			snprintf(m_lasterr, SCAP_LASTERR_SIZE, "Cannot delete session for sandbox %s", sandbox.c_str());
 			return SCAP_FAILURE;
 		}
-	}	
+	}
 
 	m_capture_started = false;
     return SCAP_SUCCESS;
 }
 
-uint32_t engine::get_vxid(uint64_t xid)
+uint32_t engine::get_vxid(uint64_t xid) const
 {
 	return parsers::get_vxid(xid);
 }
 
-int32_t engine::get_stats(scap_stats *stats)
+int32_t engine::get_stats(scap_stats *stats) const
```

Comment:
Quick question here: We sure we want cost correction here?

Reply:
Well, I did the change looking into the function and all it does is fill the passed-in `scap_stats` object reading, never writing, the data members of the `engine` object. Moreover the function is named `get_*` that is usually a signal that it won't change the object it is invoked for; it usually is odd to see the opposite, a getter that writes the object. So I went ahead and changed it.
It is true that there is another similar function, `get_stats_v2` that actually (surprisingly, IMHO) writes the object and that I could not make const. Perhaps you wonder because of that? Or there is a deeper reason, that might indicate that one day this function will have to change its object?

Reply:
stats is slated for a code (not capability) refactor, let's chat more then!

Reply:
Sure. In this case it would be `engine` that should change to force a revert of the `const`, though.
I'm ok to keep this single function non-const, if you think it's better. It won't jeopardize the essence of the PR, which is enabling client code to use a number of functions that can do so with const objects. Today this is not possible.

Reply:
We will get it in a better more consistent state / style soon. As said will ping you. Again thanks for helping cleaning up and enforcing some style!